packages/schematics/src/command-line/workspace-schematic.ts

```diff
@@ -1,24 +1,47 @@
+import {
+  JsonObject,
+  logging,
+  normalize,
+  schema,
+  virtualFs
+} from '@angular-devkit/core';
+import { createConsoleLogger, NodeJsSyncHost } from '@angular-devkit/core/node';
+import {
+  SchematicEngine,
+  UnsuccessfulWorkflowExecution
+} from '@angular-devkit/schematics';
+import {
+  NodeModulesEngineHost,
+  NodeWorkflow
+} from '@angular-devkit/schematics/tools';
+import * as appRoot from 'app-root-path';
 import { execSync } from 'child_process';
 import * as fs from 'fs';
-import { readFileSync, writeFileSync, statSync } from 'fs';
-import * as path from 'path';
+import { readFileSync, statSync, writeFileSync } from 'fs';
 import { copySync, removeSync } from 'fs-extra';
-import { NodeWorkflow } from '@angular-devkit/schematics/tools';
-import { UnsuccessfulWorkflowExecution } from '@angular-devkit/schematics';
-
+import * as inquirer from 'inquirer';
+import * as path from 'path';
 import * as yargsParser from 'yargs-parser';
-import * as appRoot from 'app-root-path';
-import { virtualFs, normalize, JsonObject } from '@angular-devkit/core';
-import { NodeJsSyncHost, createConsoleLogger } from '@angular-devkit/core/node';
 
 const rootDirectory = appRoot.path;
 
 export function workspaceSchematic(args: string[]) {
   const parsedArgs = parseOptions(args);
+  const logger = createConsoleLogger(
+    parsedArgs['verbose'],
+    process.stdout,
+    process.stderr
+  );
   const outDir = compileTools();
+  if (parsedArgs['list-schematics']) {
```

Comment:
```suggestion
  if (parsedArgs.listSchematics) {
```

This should also work after altering the `parseOptions` function